airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -266,8 +277,35 @@ private void handleEmittedErrorTrace(final AirbyteTraceMessage errorTraceMessage
     }
   }
 
-  @SuppressWarnings("PMD") // until method is implemented
-  private void handleEmittedEstimateTrace(final AirbyteTraceMessage estimateTraceMessage, final ConnectorType connectorType) {
+  /**
+   * There are several assumptions here:
+   * <p>
+   * - Assume the estimate is a whole number and not a sum i.e. each estimate replaces the previous
+   * estimate.
+   * <p>
+   * - Sources cannot emit both STREAM and SYNC estimates in a same sync. Error out if this happens.
+   */
+  @SuppressWarnings("PMD.AvoidDuplicateLiterals")
+  private void handleEmittedEstimateTrace(final AirbyteEstimateTraceMessage estimate) {
+    switch (estimate.getType()) {
+      case STREAM -> {
+        Preconditions.checkArgument(totalBytesEstimatedSync == null, "STREAM and SYNC estimates should not be emitted in the same sync.");
+        Preconditions.checkArgument(totalRecordsEstimatedSync == null, "STREAM and SYNC estimates should not be emitted in the same sync.");
+
+        log.info("Saving records estimates for namespace: {}, stream: {}", estimate.getNamespace(), estimate.getName());
```

Comment:
I'd expect this to become very noisy moving forward, can we have it as debug instead?